README.md

```diff
@@ -104,3 +115,173 @@ To configure rust-analyzer in Visual Studio Code:
     "rust-analyzer.rustfmt.overrideCommand": ["nix-shell", "--run", "cargo fmt"],
 }
 ```
+
+Editor support
+==============
+
+If you are working on the Rust code only, rust-analyzer should work perfectly out of the box, though NixOS users will need to configure rust-analyzer to wrap cargo invocations (see above).
+
+But if you are working on the C++ code, editor support is only really possible in upstream SpiderMonkey (aka “mozilla-central”), but once you’ve set up your editor in your upstream checkout, you can work on your changes there, then import them here as needed for local testing.
+
+This guide assumes that your code is checked out at:
+
+* **~/code/mozjs** for this repo
+* **~/code/mozilla-unified** for upstream SpiderMonkey
+* (NixOS users only) **~/code/nixpkgs-mozilla** for [mozilla/nixpkgs-mozilla](https://github.com/mozilla/nixpkgs-mozilla)
+
+**NixOS users:** some steps have a note in \[brackets] saying they need to be wrapped in nix-shell. Those commands should be wrapped as follows:
+
+```
+nix-shell ~/code/nixpkgs-mozilla/release.nix -A gecko.x86_64-linux.clang --run '...'
+```
+
+C++ editor setup
+----------------
+
+Start by checking out mozilla-unified ([Building Firefox on Linux](https://firefox-source-docs.mozilla.org/setup/linux_build.html) §§ 1 and 2), then create your MOZCONFIG file ([Building and testing SpiderMonkey](https://firefox-source-docs.mozilla.org/js/build.html)). I recommend (and this guide assumes) that the file is named `debug.mozconfig`, because simple names like `debug` can cause MozconfigFindException problems. The file should look like this:
+
+```
+# Build only the JS shell
+ac_add_options --enable-project=js
+
+# Enable the debugging tools: Assertions, debug only code etc.
+ac_add_options --enable-debug
+
+# Enable optimizations as well so that the test suite runs much faster. If
+# you are having trouble using a debugger, you should disable optimization.
+ac_add_options --enable-optimize
+
+# Use a dedicated objdir for SpiderMonkey debug builds to avoid
+# conflicting with Firefox build with default configuration.
+mk_add_options MOZ_OBJDIR=@TOPSRCDIR@/obj-debug-@CONFIG_GUESS@
+```
+
+If you are a NixOS user, clone [mozilla/nixpkgs-mozilla](https://github.com/mozilla/nixpkgs-mozilla) next to your `mozilla-unified` checkout, and add the following line to the start of your `debug.mozconfig`:
```

Comment:
When checking out the mozilla-unified source by following the [linked](https://firefox-source-docs.mozilla.org/setup/linux_build.htm) instructions, the bootstrap command failed as Nix is it is not supported by mach. 
```
NotImplementedError: Bootstrap support for this Linux distro not yet available: nixos

  File "/home/mukilan/dev/mozilla-unified/python/mozboot/mozboot/mach_commands.py", line 43, in bootstrap
    bootstrapper = Bootstrapper(
  File "/home/mukilan/dev/mozilla-unified/python/mozboot/mozboot/bootstrap.py", line 218, in __init__
    raise NotImplementedError(
```
Is the line `ac_add_options --disable-bootstrap` in  the generated file `.mozconfig.nix-shell` file there to overcome this issue?

Should we add a note here indicating that bootstrap failure during checkout is expected on NixOS?

Reply:
i don’t know if `--disable-bootstrap` is related to bootstrap.py, but good catch! done in 5ff7e8da587a6a6e10a55aece26e9373bd46f112